src/data/zmod/basic.lean

```diff
@@ -814,13 +814,13 @@ instance subsingleton_ring_hom [semiring R] : subsingleton ((zmod n) →+* R) :=
 instance subsingleton_ring_equiv [semiring R] : subsingleton (zmod n ≃+* R) :=
 ⟨λ f g, by { rw ring_equiv.coe_ring_hom_inj_iff, apply ring_hom.ext_zmod _ _ }⟩
 
+lemma ring_hom_right_inverse [ring R] (f : R →+* (zmod n)) :
+  function.right_inverse (coe : zmod n → R) f :=
+by { intros k, cases n; simp }
+
 lemma ring_hom_surjective [ring R] (f : R →+* (zmod n)) :
   function.surjective f :=
-begin
-  intros k,
-  rcases zmod.int_cast_surjective k with ⟨n, rfl⟩,
-  refine ⟨n, f.map_int_cast n⟩
-end
+(ring_hom_right_inverse f).surjective
```

Comment:
I couldn't work out how to use the previous proof strategy here, so faux-golfed it by just using `simp`.